src/App.jsx

```diff
@@ -1,25 +1,112 @@
 import { useState } from "react";
 import "./App.css";
 
-function City(props) {
-  return <div>This is the City component</div>;
+function City({State, Lat, Long, EstimatedPopulation, TotalWages, LocationText}) {
+  
+  return (
+    <div className="card my-5">
+      <div className="card-header">
+        {LocationText}
+      </div>
+      <div className="card-body">
+        <ul className="p-8 my-3">
+          <li>State: {State}</li>
+          <li>Location: ({Lat}, {Long})</li>
+          <li>Population (estimated): {EstimatedPopulation}</li>
+          <li>Total Wages: {TotalWages}</li>
+        </ul>
+      </div>
+    </div>
+  );
 }
 
 function ZipSearchField(props) {
-  return <div>This is the ZipSearchField component</div>;
+  
+  
+  const [zipCode, setZipCode] = useState('');
+  
+  
+  
+  return (
+    <label className="d-flex align-center py-4 px-1" for="zip-code">
+      <b className="py-2 w-25">Zip Code:</b>
+      <input id="zip-code" className="form-control p-1 rounded max-w" placeholder="Try 10016" value={zipCode} onChange={(e) => {
+        
+        var newZipCode = e.target.value.replace(/\D+/g, '').slice(0, 5);
```

Comment:
```suggestion
        const newZipCode = e.target.value.replace(/\D+/g, '').slice(0, 5);
```

The days of `var` are finished. Prefer `const`.